hotsos/core/host_helpers/cli.py

```diff
@@ -219,16 +231,36 @@ def __call__(self, *args, **kwargs):
         if not os.path.exists(self.path):
             raise SourceNotFound()
 
-        # NOTE: any post-exec hooks much be aware that their input will be
+        # NOTE: any post-exec hooks must be aware that their input will be
         # defined by the following.
-        if self.safe_decode:
-            # Some content can result in UnicodeDecodeError so use
-            # surrogateescape but only seldom.
-            output = self.safe_readlines(self.path)
-        elif self.json_decode:
+        if self.json_decode:
             output = json.load(open(self.path))
         else:
-            output = open(self.path, 'r').readlines()
+            output = []
+            ln = 0
+            with open(self.path, 'rb') as fd:
+                for line in fd:
```

Comment:
Shouldn't this be fd.readlines() ?

Reply:
no because we want to do one line at a time, if you do readlines() its all or nothing.

Reply:
i.e. if you do fd.readlines() and any line is not decodable, all will fail